src/zenml/artifacts/external_artifact.py

```diff
@@ -40,29 +39,32 @@ class ExternalArtifact(ExternalArtifactConfiguration):
     can be used to provide any value as input to a step without needing to
     write an additional step that returns this value.
 
-    This class can be configured using the following parameters:
-    - value: The artifact value (any python object), that will be uploaded to
-        the artifact store.
-    - id: The ID of an artifact that is already registered in ZenML.
-    - pipeline_name & artifact_name: Name of a pipeline and artifact to search
-        in latest run.
-    - model_name & model_version & model_artifact_name & model_artifact_version:
-        Name of a model, model version, model artifact and artifact version to
-        search.
+    The external artifact needs to have either a value associated with it
+    that will be uploaded to the artifact store, or reference an artifact
+    that is already registered in ZenML.
+
+    There are several ways to reference an existing artifact:
+    - By providing an artifact ID.
+    - By providing an artifact name and version.
+    - By providing an artifact name and a pipeline run name.
+    - By providing an artifact name and a pipeline name, in which case ZenML
+        will attempt to find the artifact in the latest run of the pipeline.
+    - By only providing an artifact name, in which case the latest version
+        of the artifact will be used.
 
     Args:
         value: The artifact value.
         id: The ID of an artifact that should be referenced by this external
             artifact.
-        pipeline_name: Name of a pipeline to search for artifact in latest run.
-        artifact_name: Name of an artifact to be searched in latest pipeline
-            run.
-        model_name: Name of a model to search for artifact in (if None -
-            derived from step context).
-        model_version: Version of a model to search for artifact in (if None -
-            derived from step context).
-        model_artifact_name: Name of a model artifact to search for.
-        model_artifact_version: Version of a model artifact to search for.
+        name: Name of an artifact to search. If none of
+            `version`, `pipeline_run_name`, or `pipeline_name` are set, the
+            latest version of the artifact will be used.
+        version: Version of the artifact to search. Only used when `name` is
+            provided.
+        pipeline_run_name: Name of a pipeline run to search artifacts in. Only
+            used when `name` is provided.
+        pipeline_name: Name of a pipeline in which to search for the artifact.
+            Only used when `name` is provided.
```

Comment:
Conceptually, I'm not getting why we are giving privilege to use pipeline objects as selectors, but not model versions? Pipeline run name to me is somewhat of the same level of granularity as a model version.

Reply:
Fair enough - I just added it because we had it in the previous version but I'd be happy to take them out as well

Reply:
Let's kick them out, if not needed. But, if we want to merge this - don't change it now, just make a follow-up PR. It would be better.

Reply:
Ok done 👍